File: Interview/面试题01.05.一次编辑.py
"""
字符串有三种编辑操作:插入一个字符、删除一个字符或者替换一个字符。 
给定两个字符串，编写一个函数判定它们是否只需要一次(或者零次)编辑。
"""


class Solution:
    def oneEditAway(self, first: str, second: str) -> bool:
        if abs(len(first) - len(second)) > 1:
            return False

        i, j = 0, 0
        c = 0

        while i < len(first) and j < len(second):
            if first[i] == second[j]:
                i += 1
                j += 1
            elif True:
                pass
        pass


if __name__ == "__main__":
    print(Solution().oneEditAway('pale', 'ple'))
    print(Solution().oneEditAway('pales', 'pal'))

File: LeetCode/gold_book/面试题 01.05. 一次编辑.cpp
#include <algorithm>
#include <string>

using namespace std;

class Solution {
  public:
    bool oneEditAway(string first, string second) {
        int dp[ first.size() ][ second.size() ];
        memset(dp, 0xFFFFFF, sizeof(dp));

        for (int i = 0; i < first.size(); i++) {
            for (int j = 0; j < second.size(); j++) {

            }
        }

        return true;
    }
};